gateway/src/apicast/policy/token_introspection/token_introspection.lua

```diff
@@ -77,6 +85,12 @@ local function introspect_token(self, token)
 end
 
 function _M:access(context)
+  if self.auth_type == "oidc_issuer_endpoint" then
+    local components = resty_url.parse(context.service.oidc.issuer_endpoint)
+    self.credential = create_credential(components.user, components.password)
+    self.introspection_url = context.proxy.oauth.config.openid.token_introspection_endpoint
```

Comment:
We probably should test how this policy behaves on non OIDC service. I think this would crash.